Fixes to achieve compatibility with mlr3 weights update (#914)
mlr-org/mlr3pipelines: NEWS.md

* Added missing error for predicting with untrained `PipeOp`s / `Graph`s.
* Fix: Corrected typo in the hyperparameter name `use_parallel` of `PipeOpVtreat`.

> * Fix: New hyperparameter of `bbotk::OptimizerBatchNLoptr` does not get overwritten in `LearnerClassifAvg` / `LearnerRegrAvg`'s internal `optimize_weights_learneravg` function.

Reviewer: ```suggestion
* Fix: Do not overwrite initial hyperparameter settings of `bbotk::OptimizerBatchNLoptr` in `LearnerClassifAvg` / `LearnerRegrAvg`'s internal `optimize_weights_learneravg` function.
```
